app/cypress/support/commands.js

```diff
@@ -640,3 +653,36 @@ Cypress.Commands.add(
       .should("have.text", status);
   }
 );
+
+// TODO: possible candidate to replace other date related commands
+Cypress.Commands.add("setDate", (date, ariaLabel, reportNumber = 0) => {
```

Comment:
for a better flowing API, we should be more precise in the name and invert the arguments:
```suggestion
Cypress.Commands.add("setDateInPicker", (ariaLabel, date, reportNumber = 0) => {
```

this way we can call it with something like
```js
cy.setDateInPicker("Contract Start Date", "2022-01-01");
```